src/aws/credential.rs

```diff
@@ -452,7 +456,7 @@ fn canonicalize_headers(header_map: &HeaderMap) -> (String, String) {
             if value_idx != 0 {
                 canonical_headers.push(',');
             }
-            canonical_headers.push_str(value.trim());
+            canonical_headers.push_str(&normalize_whitespace(value));
```

Comment:
It seems rather wasteful to allocate both a Vec and a String only to append there contents to another string.

Perhaps we could just do something like (not tested)

```
let mut iter = value.trim().split(' ').filter(|x| !x.is_empty());
if let Some(first) = iter.next() {
    canonical_headers.push(first);
    for x in iter {
       canonical_headers.push(' ');
       canonical_headers.push(x);
    }
}
```